src/java.base/share/classes/java/net/SocketPermission.java

```diff
@@ -694,6 +700,15 @@ private static String checkForIDN(String name) {
         }
     }
 
+    private boolean matchArr(String[] cnames, String hname) {
+        for (String cname : cnames) {
+            if (Objects.nonNull(cname) && match(cname, hname)) {
+                return true;
```

Comment:
Sorry, forgot to mention, I think I'd prefer just to have:
if (cname != null && match(..)))

I think Objects.nonNull exists more for use as a method reference.

Reply:
Done.